File: HealthAxis.Admin/Services/TokenService.cs
using Microsoft.JSInterop;

namespace HealthAxis.Admin.Services
{
    public class TokenService
    {
        private readonly IJSRuntime _jsRuntime;

        private const string AccessTokenKey = "healthaxis_access_token";
        private const string RefreshTokenKey = "healthaxis_refresh_token";
        private const string UserRoleKey = "healthaxis_user_role";
        private const string UserEmailKey = "healthaxis_user_email";
        private const string UserIdKey = "healthaxis_user_id";

        private const string LocalStorageSetItem = "localStorage.setItem";
        private const string LocalStorageGetItem = "localStorage.getItem";
        private const string LocalStorageRemoveItem = "localStorage.removeItem";

        public TokenService(IJSRuntime jsRuntime)
        {
            _jsRuntime = jsRuntime;
        }

        public async Task SaveTokensAsync(
            string accessToken,
            string refreshToken,
            string role,
            string email,
            string userId)
        {
            await _jsRuntime.InvokeVoidAsync(
                LocalStorageSetItem,
                AccessTokenKey,
                accessToken);

            await _jsRuntime.InvokeVoidAsync(
                LocalStorageSetItem,
                RefreshTokenKey,
                refreshToken);

            await _jsRuntime.InvokeVoidAsync(
                LocalStorageSetItem,
                UserRoleKey,
                role);

            await _jsRuntime.InvokeVoidAsync(
                LocalStorageSetItem,
                UserEmailKey,
                email);

            await _jsRuntime.InvokeVoidAsync(
                LocalStorageSetItem,
                UserIdKey,
                userId);
        }

        public async Task<string?> GetAccessTokenAsync()
        {
            return await _jsRuntime.InvokeAsync<string?>(
                LocalStorageGetItem,
                AccessTokenKey);
        }

        public async Task<string?> GetUserRoleAsync()
        {
            return await _jsRuntime.InvokeAsync<string?>(
                LocalStorageGetItem,
                UserRoleKey);
        }

        public async Task ClearTokensAsync()
        {
            await _jsRuntime.InvokeVoidAsync(
                LocalStorageRemoveItem,
                AccessTokenKey);

            await _jsRuntime.InvokeVoidAsync(
                LocalStorageRemoveItem,
                RefreshTokenKey);

            await _jsRuntime.InvokeVoidAsync(
                LocalStorageRemoveItem,
                UserRoleKey);

            await _jsRuntime.InvokeVoidAsync(
                LocalStorageRemoveItem,
                UserEmailKey);

            await _jsRuntime.InvokeVoidAsync(
                LocalStorageRemoveItem,
                UserIdKey);
        }
    }
}

File: HealthAxis.Angular/src/app/core/models/token.service.ts
import { Injectable } from '@angular/core';
import { AuthResponse } from '../models/auth-response';

@Injectable({
  providedIn: 'root'
})
export class TokenService {
  private readonly accessTokenKey = 'healthaxis_access_token';
  private readonly refreshTokenKey = 'healthaxis_refresh_token';
  private readonly userRoleKey = 'healthaxis_user_role';
  private readonly userEmailKey = 'healthaxis_user_email';
  private readonly userIdKey = 'healthaxis_user_id';
  private readonly referenceIdKey = 'healthaxis_reference_id';

  saveAuthData(response: AuthResponse): void {
    localStorage.setItem(this.accessTokenKey, response.accessToken);
    localStorage.setItem(this.refreshTokenKey, response.refreshToken);
    localStorage.setItem(this.userRoleKey, response.role);
    localStorage.setItem(this.userEmailKey, response.email);
    localStorage.setItem(this.userIdKey, response.userId);
    localStorage.setItem(this.referenceIdKey, String(response.referenceId));
  }

  getAccessToken(): string | null {
    return localStorage.getItem(this.accessTokenKey);
  }

  getRefreshToken(): string | null {
    return localStorage.getItem(this.refreshTokenKey);
  }

  getUserRole(): string | null {
    return localStorage.getItem(this.userRoleKey);
  }

  getUserEmail(): string | null {
    return localStorage.getItem(this.userEmailKey);
  }

  getUserId(): string | null {
    return localStorage.getItem(this.userIdKey);
  }

  getReferenceId(): string | null {
    return localStorage.getItem(this.referenceIdKey);
  }

  isLoggedIn(): boolean {
    return !!this.getAccessToken();
  }

  clearAuthData(): void {
    localStorage.removeItem(this.accessTokenKey);
    localStorage.removeItem(this.refreshTokenKey);
    localStorage.removeItem(this.userRoleKey);
    localStorage.removeItem(this.userEmailKey);
    localStorage.removeItem(this.userIdKey);
    localStorage.removeItem(this.referenceIdKey);
  }
}